dll/win32/shell32/droptargets/CFSDropTarget.cpp

```diff
@@ -47,6 +47,17 @@ HRESULT CFSDropTarget::_CopyItems(IShellFolder * pSFFrom, UINT cidl,
     /* Build a double null terminated list of C strings from source paths */
     for (UINT i = 0; i < cidl; i++)
     {
+        SFGAOF rfg = SFGAO_FILESYSANCESTOR | SFGAO_FILESYSTEM;
+        ret = pSFFrom->GetAttributesOf(1, apidl[i], &rfg);
+        if (FAILED(ret))
+            goto cleanup;
+
+        if (!(rfg & (SFGAO_FILESYSANCESTOR | SFGAO_FILESYSTEM)))
+        {
+            WARN("Skipping item without SFGAO_FILESYSANCESTOR or SFGAO_FILESYSTEM\n");
+            continue;
```

Comment:
This whole loop should be moved to DragEnter so the entire operation is forced to drop effect link if there are non-food system items in the data object? What does Windows do?

Reply:
I'm ok with keeping this check like that to be sure that an application doesn't do something crazy. Your idea that DragEnter can filter the effects makes sense but it needs to be checked if windows does the same. If we add such a filtering we should also add a loud warning that this is unexpected. I wonder if this is where the wrong flag comes from:

https://github.com/reactos/reactos/blob/master/dll/win32/shell32/CDefView.cpp#L2214

perhaps we should begin with 0 and if the folder doesn't like any of the copy , move , or link operations we should stop the drag and drop?